numpy/core/_dtype_ctypes.py

```diff
@@ -61,12 +60,24 @@ def _from_ctypes_structure(t):
     else:
         fields = []
         for fname, ftyp in t._fields_:
-            fields.append((fname, dtype_from_ctypes_type(ftyp)))
+            fields.append((fname, dtype_from_ctypes_endianness(ftyp)))
 
         # by default, ctypes structs are aligned
         return np.dtype(fields, align=True)
 
 
+def dtype_from_ctypes_endianness(t):
+    """
+    Return the dtype type with endianness included if it's the case
+    """
+    if t.__name__.endswith('_be'):
```

Comment:
Found something better - `if t.__ctypes_be__ is t`

Reply:
I don't follow. 
The ctype type has both `_be` and `_le` in it's `__dict__ `independent of actual endianness.